pyOCD/tools/flash_tool.py

```diff
@@ -165,7 +205,7 @@ def main():
                                  fast_verify=args.fast_program)
 
             # Intel hex file format
-            if args.format == 'hex':
+            elif args.format == 'hex':
                 hex = IntelHex(args.file)
                 addresses = hex.addresses()
                 addresses.sort()
```

Comment:
Good idea on the auto file type detection.  You might also want to have it print a warning if the file type could not be determined.